src/Paramore.Brighter.Outbox.DynamoDB/DynamoDbOutbox.cs

```diff
@@ -400,13 +471,10 @@ public async Task MarkDispatchedAsync(
         /// <param name="args"></param>
         public void MarkDispatched(string id, RequestContext requestContext, DateTimeOffset? dispatchedAt = null, Dictionary<string, object> args = null)
         {
-            var message = _context.LoadAsync<MessageItem>(id, _dynamoOverwriteTableConfig).Result;
-            MarkMessageDispatched(dispatchedAt ?? _timeProvider.GetUtcNow(), message);
-
-            _context.SaveAsync(
-                message, 
-                _dynamoOverwriteTableConfig)
-                .Wait(_configuration.Timeout);
+            MarkDispatchedAsync(id, requestContext, dispatchedAt, args)
+                .ConfigureAwait(ContinueOnCapturedContext)
```

Comment:
I wonder if we should use Brighter's AsyncContext and not GetResult()

Reply:
We use `GetAwaiter().GetResult()` in a lot of places throughout the codebase - if that's not the way we should be doing things then it should probably be in a separate PR that does it everywhere.

Reply:
I did use a PR to try and fix many of them, as part of Proactor/Reactor work. I may well not have found all of them. Let's drop for your issue, and I will pick this up along with any others that have remained or crept back in